### PR TITLE
Fix empty values in older (pre-3.0) models

### LINKED ISSUE #3829: Model fails to load if value in storage.get_specification_value_from_value is None
### Describe the bug
Opening an existing model failed at line 633 of storage.py, where `value `is an empty string.

```
 File "/Users/mikekidner/BitBucketRepos/gaphor/gaphor/storage/storage.py", line 620, in upgrade_simple_properties_to_value_specifications
    defaultValue = get_value_specification_from_value(
        value, elem, element_factory, modeling_language
    )
  File "/Users/mikekidner/BitBucketRepos/gaphor/gaphor/storage/storage.py", line 633, in get_value_specification_from_value
    if value[0] == '"' and value[-1] == '"':
       ~~~~~^^^
IndexError: string index out of range

```

### Expected behavior
Valid models from earlier versions  should load

### OS
- [ ] Linux (Please put in notes the specific distro)
- [x ] macOS
- [ ] Windows

### Version
Version of Gaphor:3.0

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [x] Bug fix
- [ ] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

Deal with odd cases, e.g. when value of a value is empty.

Issue Number: Fixes #3829

### What is the new behavior?

* Added regression tests for ValueSpecification update.
* Added and fixed corner cases (superseeds #3830)
* Type all storage functions. This will reduce the room for error in the future.
* Cleanup: no longer need to deal with "homeless literals"

### Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by mikekidner** [comment]
Thanks @amolenaar !

**Review by danyeaw — APPROVED** [review]
Hi @amolenaar, this is a really nice fix and cleanup! Just one nitpick about a comment.

File: gaphor/storage/parser.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ class base:
 
     def __init__(self):
         # added Base as an alternate type for values to support upgrading simple types to UML.ValueSpecification
```
Comment (danyeaw) [inline gaphor/storage/parser.py:43]:
```suggestion
```

**Comment by amolenaar** [comment]
Fixed.
